### DISCUSSION TITLE
Why need to upgrade rust-version at the same time when upgrading the fd version?

### DISCUSSION THREAD

**Discussion by Bin-ly** [sharkdp/fd#1561] [Q&A]
The code difference between fd9.0.0 version and fd10.1.0 version does not seem to be very big (I have not yet studied whether the adjusted code uses the functions provided by higher version dependencies), but some package dependencies and
The rust version has been upgraded (1.70.0->1.77.2). What is the reason for this?
Without extensive adjustments, my rust1.70.0 can also compile fd10.1.0 without requiring me to re-download rust1.77.2, which is more convenient for me.

**Comment by tmccombs** [comment]
Because of https://blog.rust-lang.org/2024/04/09/cve-2024-24576.html

If you aren't on windows, and it compiles with 1.70, then sure, compile it with 1.70.
